lib/grinder_files.dart

```diff
@@ -98,12 +100,210 @@ class FileSet {
 }
 
 /**
+ * A class to make it easier to manipulate file system entites. Once paths or
+ * entites are converted into `Path`s, they can be easily copied, deleted,
+ * joined, and their name retrieved.
+ */
+class Path {
+  /**
+   * Creates a temporary directory in the system temp directory. See
+   * [Directory.systemTemp] and [Directory.createTempSync]. If [prefix] is
+   * missing or null, the empty string is used for [prefix].
+   */
+  static Path createSystemTemp([String prefix]) {
+    return new Path(Directory.systemTemp.createTempSync(prefix));
+  }
+
+  static Path get cwd => new Path(Directory.current);
+
+  final String _path;
+
+  /**
+   * Create a new [Path]. The [entity_or_path] parameter can be a
+   * [FileSystemEntity] or a [String].
+   */
+  Path(entity_or_path) : _path = _coerce(entity_or_path);
+
+  String get name {
+    int index = _path.lastIndexOf(_sep);
+    return index != -1 ? _path.substring(index + 1) : null;
+  }
+
+  String get path => _path;
+
+  FileSystemEntity get entity {
+    final FileSystemEntityType type = FileSystemEntity.typeSync(_path);
+
+    if (type == FileSystemEntityType.FILE) {
+      return new File(_path);
+    } else if (type == FileSystemEntityType.DIRECTORY) {
+      return new Directory(_path);
+    } else if (type == FileSystemEntityType.LINK) {
+      return new Link(_path);
+    } else {
+      return null;
+    }
+  }
+
+  bool get exists {
+    return FileSystemEntity.typeSync(_path) != FileSystemEntityType.NOT_FOUND;
+  }
+
+  /**
+   * Returns the containing [Path]. Returns a non-null value even if this is a
+   * root directory.
+   *
+   * See [FileSystemEntity.parent].
+   */
+  Path get parent {
+    int index = _path.lastIndexOf(_sep);
+
+    // Do string manipulation if there are path separators; otherwise, use the
+    // file system entity information.
+    if (index == 0 || index == -1) {
+      FileSystemEntity e = entity;
+      return e == null ? null : new Path(e.parent);
+    } else {
+      return new Path(_path.substring(0, index));
+    }
+  }
+
+//  /**
+//   * Returns the abolute version of this Path.
+//   */
+//  Path get absolute {
+//    // TODO:
+//  }
+
+  bool get isDirectory => FileSystemEntity.isDirectorySync(_path);
+  bool get isFile => FileSystemEntity.isFileSync(_path);
+  bool get isLink => FileSystemEntity.isLinkSync(_path);
+
+  /**
```

Comment:
Per #102 can we make this /// ?  Or we can just wait until the linter supports batch fixes.